doc/API.md

```diff
@@ -777,6 +745,37 @@ const peerStats = libp2p.metrics.forProtocol('/meshsub/1.0.0')
 console.log(peerStats.toJSON())
 ```
 
+## Events
+
+Once you have a libp2p instance, you are able to listen to several events it emits, so that you can be noticed of relevant network events.
+
+#### An error has occurred
+
+`libp2p.on('error', (err) => {})`
+
+- `err`: instance of `Error`
+
+#### A peer has been discovered
+
+`libp2p.on('peer:discovery', (peer) => {})`
+
+If `autoDial` option is `true`, applications should **not** attempt to connect to the peer
+unless they are performing a specific action. See [peer discovery and auto dial](./PEER_DISCOVERY.md) for more information.
+
+- `peer`: instance of [PeerInfo][https://github.com/libp2p/js-peer-info]
+
+#### We have a new connection to a peer
+
+`libp2p.on('peer:connect', (peer) => {})`
+
+- `peer`: instance of [PeerInfo][https://github.com/libp2p/js-peer-info]
+
+#### We have closed a connection to a peer
```

Comment:
```suggestion
#### An existing connection to a peer has been closed
```

Can we explain why this might happen? Is it only if the remote disconnects or do we also get this event when we disconnect?

Reply:
It occurs anytime we are disconnected from a peer, regardless of what caused it.